hardware/evohome.cpp

```diff
@@ -173,15 +178,42 @@ bool CEvohome::StartHardware()
 		m_retrycntr=RETRY_DELAY; //will force reconnect first thing
 		
 		std::vector<std::vector<std::string> > result;
-		result = m_sql.safe_query("SELECT Name,DeviceID,nValue FROM DeviceStatus WHERE (HardwareID==%d) AND (Unit==0)", m_HwdID);
+		result = m_sql.safe_query("SELECT Name,DeviceID,nValue FROM DeviceStatus WHERE (HardwareID==%d) AND (Unit==0) AND (Type==%d)", m_HwdID, (int)pTypeEvohome);
+			
 		if (result.size()>0)
 		{
 			std::vector<std::string> sd=result[0];
 			std::stringstream s_strid;
 			s_strid << std::hex << sd[1];
-			s_strid >> m_nDevID;
+			if (s_strid.str().size() != 0)  // Check whether the controller's DeviceID has been set
```

Comment:
Thanks for the feedback.  I'll switch to using .empty() in the code and re-test.